tests/test_newline.py

```diff
@@ -1,21 +1,14 @@
-import sys
-import unicodedata
-import re
-from glob import *
+import glob
 import unittest
 
+PROTO_FILES = glob.glob("*.proto")
 
 class TestNewLine(unittest.TestCase):
-    ''' Test class for mandatory new line. ''' 
+    ''' Test class for mandatory new line. '''
 
     def test_newline(self):
         ''' Test to check last line of file must end with a new line. '''
-        for file in glob("*.proto"):
-            with open(file, "rt") as fin:
-                hasNewLine = True
-
-                for line in fin:
-                    hasNewLine = line.endswith("\n")
-
-                self.assertTrue(hasNewLine, file + " has no new line at the end of the file.")
-                
\ No newline at end of file
+        for file in PROTO_FILES:
+            with open(file, "rt") as fin, self.subTest(file=file):
+                lastCharacter = fin.read()[-1]
+                self.assertEqual(lastCharacter, "\n")
```

Comment:
Please add `self.assertEqual(lastCharacter, "\n", file + " has no new line at the end of the file")` to excplictly warn the user that he is missing a newline character.

Reply:
Done! Thanks for the review!